### ISSUE TITLE
Dynamo backend custom settings

### ISSUE THREAD

**Issue by pratogab** [huggingface/accelerate#1978]
### System Info

```Shell
accelerate v.0.22.0
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

Setting true to either `dynamo_use_dynamic` or `dynamo_use_fullgraph` results in an error in my accelerate config, with dynamo_backend=INDUCTOR and any dynamo_mode (default, reduce-overhead or max-autotune):
```
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: 'NO'
downcast_bf16: 'no'
dynamo_config:
  dynamo_backend: INDUCTOR
  dynamo_mode: default
  dynamo_use_dynamic: true
  dynamo_use_fullgraph: true
gpu_ids: '0'
machine_rank: 0
main_training_function: main
mixed_precision: bf16
num_machines: 1
num_processes: 1
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

The errors vary depending on the dynamo_mode being used, so I won't print all of them here, but as an example, the prior config results in:
```
Traceback (most recent call last):
  File "/home/example_script.py", line 314, in <module>
    main()                                                                                   
  File "/home/example_script.py", line 310, in main
    run(accelerator, start_time, args)                                                                                                       
  File "/home/example_script.py", line 76, in run
    train_loss = train_epoch(accelerator, model, train_dataloader, optimizer, lr_scheduler, criterion, device, args)                            
  File "/home/example_script.py", line 125, in train_epoch
    output = model(input_ids=minibatch, attention_mask=mini_attention_mask)                                                                      
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)                                                                                                                                                                                                       File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 82, in forward                                                                                                           return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)                                                                                                        
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 209, in _fn
    return fn(*args, **kwargs)                                 
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 337, in catch_errors
    return callback(frame, cache_size, hooks)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 104, in _fn
    return fn(*args, **kwargs)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 262, in _convert_frame_assert
    return _compile(
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 163, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 324, in _compile
    out_code = transform_code_object(code, transform)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 445, in transform_code_object
    transformations(instructions, code_options)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 311, in transform
    tracer.run()
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1726, in run
    super().run()
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 576, in run
    and self.step()
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 540, in step
    getattr(self, inst.opname)(inst)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 342, in wrapper
    return inner_fn(self, inst)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1002, in CALL_FUNCTION_EX
    self.call_function(fn, argsvars.items, kwargsvars.items)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 474, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/variables/user_defined.py", line 264, in call_function
    return super().call_function(tx, args, kwargs)
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/variables/base.py", line 230, in call_function
    unimplemented(f"call_function {self} {args} {kwargs}")
  File "/home/miniconda3/envs/hf-torch2/lib/python3.10/site-packages/torch/_dynamo/exc.py", line 71, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: call_function UserDefinedObjectVariable(ConvertOutputsToFp32) [NNModuleVariable()] OrderedDict([('input_ids', TensorVariable()), ('attention_mask', TensorVariable())])
```

### Expected behavior

Unsure what the error is, or the settings shouldn't be available if they aren't compatible with the inductor backend.

**Comment by SunMarc** [comment]
Hi @pratogab , thanks for reporting. Can you provide a minimal reproducer ?

**Comment by pratogab** [comment]
The following on my laptop crashes (CPU only). Also tried the same on a cluster with GPU and it also crashes. You can try different combinations for the torch dynamo settings, it gives a different error. To get no errors, both `dynamo_use_dynamic` and `dynamo_use_fullgraph` have to be set to false.
```yaml
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: 'NO'
downcast_bf16: 'no'
dynamo_config:
  dynamo_backend: INDUCTOR
  dynamo_mode: default
  dynamo_use_dynamic: true
  dynamo_use_fullgraph: true
ipex_config:
  ipex: false
machine_rank: 0
main_training_function: main
mixed_precision: 'no'
num_machines: 1
num_processes: 1
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: true
```

```python
import torch
from accelerate import Accelerator
from transformers import AutoTokenizer, AutoModelForCausalLM


def main():
    accelerator = Accelerator()

    device = accelerator.device

    model = AutoModelForCausalLM.from_pretrained('facebook/opt-125m').to(device)
    tokenizer = AutoTokenizer.from_pretrained('facebook/opt-125m')
    optimizer = torch.optim.AdamW(model.parameters())

    model, optimizer = accelerator.prepare(model, optimizer)

    batch = tokenizer(['I like turtles.'], return_tensors='pt')
    output = model(**batch)


if __name__ == '__main__':
    main()
```

**Comment by pratogab** [comment]
This issue seems to be fixed in v0.23 of accelerate. Thank you.
